### PR TITLE
Web console: add log tailing to task log view 

### PR DESCRIPTION
**Pull request by mcbrewster**
Screen Shot 2019-05-20 at 1 15 06 PM

Adds a check box to the task log view. When the check box is checked the get log function is called every 2 seconds and updates the log as well as position the scroll bar at the bottom.

### CODE REVIEW

**Comment by vogievetsky** [comment]
I think you should move 
image
to be left aligned

File: web-console/src/components/show-log.tsx [hunk 1]
```diff
@@ -71,12 +73,39 @@ export class ShowLog extends React.Component<ShowLogProps, ShowLogState> {
...
+  private tail() {
+      this.getLogInfo().then(() => {
+        if (this.state.tail) {
+          if (this.log.current) {
+            this.log.current.scrollTo(0, this.log.current.scrollHeight); }
+          setTimeout(function() {
+            this.tail();
+          }.bind(this), 2000);
+        }
+      });
```
Comment (vogievetsky) [inline web-console/src/components/show-log.tsx]: oh my the indentation in this function is all messed up also you do not need to do `.bind(this)` you can just use `() => {}` syntax

File: web-console/src/components/show-log.tsx [hunk 2]
```diff
@@ -43,15 +43,17 @@ export interface ShowLogProps extends React.Props<any> {
 
 export interface ShowLogState {
   logValue: string;
+  tail: boolean;
 }
 
 export class ShowLog extends React.Component<ShowLogProps, ShowLogState> {
+  log = React.createRef<HTMLTextAreaElement>();
```
Comment (vogievetsky) [inline web-console/src/components/show-log.tsx]: could you denote with with a `public` prefix and also add a empty line after this one?

File: web-console/src/components/show-log.tsx [hunk 3]
```diff
@@ -71,12 +73,39 @@ export class ShowLog extends React.Component<ShowLogProps, ShowLogState> {
     }
   }
 
+  private tail() {
+      this.getLogInfo().then(() => {
```
Comment (vogievetsky) [inline web-console/src/components/show-log.tsx]: don't use `.then` use `async` / `await` instead

Comment (mcbrewster) [reply web-console/src/components/show-log.tsx]: changed

File: web-console/src/components/show-log.tsx (hunk 2 above)
Comment (mcbrewster) [reply web-console/src/components/show-log.tsx]: changed

File: web-console/src/components/show-log.tsx (hunk 1 above)
Comment (mcbrewster) [reply web-console/src/components/show-log.tsx]: changed

File: web-console/src/components/show-log.tsx (hunk 2 above)
Comment (mcbrewster) [reply web-console/src/components/show-log.tsx]: Screen Shot 2019-05-20 at 5 08 18 PM

Ts lint isn't onboard with that, should I do it anyway

File: web-console/src/components/show-log/show-log.tsx [hunk 4]
```diff
@@ -71,12 +74,39 @@ export class ShowLog extends React.Component<ShowLogProps, ShowLogState> {
...
+
+
   render() {
     const { endpoint, downloadFilename } = this.props;
     const { logValue } = this.state;
 
     return <div className="show-log">
       <div className="top-actions">
+        <Checkbox
+          label="Tail Log"
```
Comment (vogievetsky) [inline web-console/src/components/show-log/show-log.tsx]: please use "Sentence case"

**Comment by vogievetsky** [comment]
Looks good to go 👍

**Pull request merged**